### PR TITLE
Add documentation for cakeissues.net

### PR DESCRIPTION
**Pull request by pascalberger**
Add documentation for cakeissues.net, moved from Cake.Issues.Website repository including full history.

Part of https://github.com/cake-contrib/Cake.Issues.Website/issues/123

### CODE REVIEW

File: docs/examples/azure-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                        repoRootFolder);
+                }
+            }
+        }
+});
+```
+
+[TfsPullRequests]: ../../../../api/Cake.Issues.PullRequests.Tfs/TfsPullRequestSystemAliases/BC3F9B2C
+[Allow scripts to access the OAuth token]: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/options?view=vsts&tabs=yaml#allow-scripts-to-access-the-oauth-token
+[OAuth authentication from Azure Pipelines]: ../setup#oauth-authentication-from-azure-pipelines
```
Comment (x-jokay) [inline docs/examples/azure-pipelines.md]: [MD047](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md047---files-should-end-with-a-single-newline-character) markdown files should end with a single newline character.

File: docs/examples/pullrequest-id.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        TfsPullRequests(
+            repoRemoteUrl,
+            pullRequestId,
+            TfsAuthenticationNtlm()),
+        repoRootFolder);
+});
+```
+
+[TfsPullRequests]: ../../../../api/Cake.Issues.PullRequests.Tfs/TfsPullRequestSystemAliases/BC3F9B2C
+[Cake.Git]: https://www.nuget.org/packages/Cake.Git/
```
Comment (x-jokay) [inline docs/examples/pullrequest-id.md]: [MD047](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md047---files-should-end-with-a-single-newline-character) markdown files should end with a single newline character.

File: docs/features.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+For detailed instructions how to connect using the different methods see [Setup instructions].
+
+![Cake.Issues.PullRequests.Tfs](cake.issues.pullrequests.tfs.png "Cake.Issues.PullRequests.Tfs")
+
+[demo repository]: https://dev.azure.com/pberger/Cake.Issues-Demo
+[Cake.Issues.PullRequests.Tfs addin]: https://www.nuget.org/packages/Cake.Issues.PullRequests.Tfs
+[Azure DevOps]: https://azure.microsoft.com/en-us/services/devops/
+[Core features]: ../../overview/features#supported-core-functionality
+[Setup instructions]: setup
```
Comment (x-jokay) [inline docs/features.md]: [MD047](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md047---files-should-end-with-a-single-newline-character) markdown files should end with a single newline character.

File: docs/requirements.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+Order: 10
+Title: Requirements
+Description: Requirements for the Cake.Issues.PullRequests.Tfs addin.
+---
+The requirements for using the [Cake.Issues.PullRequests.Tfs addin] are listed in the [release notes] for any specific version.
+
+[Cake.Issues.PullRequests.Tfs addin]: https://www.nuget.org/packages/Cake.Issues.PullRequests.Tfs
+[release notes]: release-notes
```
Comment (x-jokay) [inline docs/requirements.md]: [MD047](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md047---files-should-end-with-a-single-newline-character) markdown files should end with a single newline character.

File: docs/setup.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+[Cake.Issues.PullRequests.Tfs addin]: https://www.nuget.org/packages/Cake.Issues.PullRequests.Tfs
+[Cake.Tfs addin]: https://www.nuget.org/packages/Cake.Tfs
+[Configure TFS to use Basic Authentication]: https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/auth/tfs-basic-auth?view=tfs-2018#configure-tfs-to-use-basic-authentication
+[Authenticate access with personal access tokens for Azure DevOps Services and TFS]: https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=vsts
+[Allow scripts to access the OAuth token]: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/options?view=vsts&tabs=yaml#allow-scripts-to-access-the-oauth-token
+[TfsAuthenticationNtlm]: https://cakebuild.net/api/Cake.Tfs/TfsAliases/6989592E
+[TfsAuthenticationBasic]: https://cakebuild.net/api/Cake.Tfs/TfsAliases/86407B1D
+[TfsAuthenticationPersonalAccessToken]: https://cakebuild.net/api/Cake.Tfs/TfsAliases/0E4AC3E3
+[TfsAuthenticationOAuth]: https://cakebuild.net/api/Cake.Tfs/TfsAliases/B5D45B5D
+[TfsAuthenticationAzureActiveDirectory]: https://cakebuild.net/api/Cake.Tfs/TfsAliases/0E787800
```
Comment (x-jokay) [inline docs/setup.md]: [MD047](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md047---files-should-end-with-a-single-newline-character) markdown files should end with a single newline character.

File: docs/features.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+# Basic features
+
+* Writes issues as comments to Team Foundation Server (TFS) or [Azure DevOps] pull requests.
+* Identification of pull requests through source branch or pull request ID.
+* Comments written by the addin will be rendered with a specific icon corresponding to the state of the issue.
+* Adds rule number and, if provided by the issue provider, link to the rule description to the comment.
+* Support for issues messages formatted in Markdown format.
+
+# Supported capabilities
```
Comment (x-jokay) [inline docs/features.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/features.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+The [Cake.Issues.PullRequests.Tfs addin] supports all [Core features].
+
+|                                                                    | Capability                     | Remarks                        |
+|--------------------------------------------------------------------|--------------------------------|--------------------------------|
+| <span class="glyphicon glyphicon-ok" style="color:green"></span>   | Checking commit ID             |                                |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span>   | Discussion threads             |                                |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span>   | Filtering by modified files    |                                |
+
+# Supported authentication methods
```
Comment (x-jokay) [inline docs/features.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/setup.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+:::{.alert .alert-info}
+NTLM authentication is only available for on-premise Team Foundation Server.
+:::
+
+To authenticate with NTLM you can use the [TfsAuthenticationNtlm] alias from the [Cake.Tfs addin].
+
+The user needs to have `Contribute to pull requests` permission for the specific repository to
+allow [Cake.Issues.PullRequests.Tfs addin] to post issues as comments to pull requests.
+
+# Basic authentication
```
Comment (x-jokay) [inline docs/setup.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/setup.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+Basic authentication is only available for on-premise Team Foundation Server.
+:::
+
+To authenticate with basic authentication you can use the [TfsAuthenticationBasic] alias from the [Cake.Tfs addin] and 
+need to [Configure TFS to use Basic Authentication].
+
+The user needs to have `Contribute to pull requests` permission for the specific repository to
+allow [Cake.Issues.PullRequests.Tfs addin] to post issues as comments to pull requests.
+
+# Personal access token
```
Comment (x-jokay) [inline docs/setup.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/setup.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+To authenticate with an personal access token you can use the [TfsAuthenticationPersonalAccessToken] alias from the [Cake.Tfs addin].
+
+If you want to use the [Cake.Issues.PullRequests.Tfs addin] with an personal access token see
+[Authenticate access with personal access tokens for Azure DevOps Services and TFS] for instructions how to create
+a personal access token.
+
+The access token needs to have the scope `Code (read and write)` set and the user needs to have `Contribute to pull requests`
+permission for the specific repository to allow [Cake.Issues.PullRequests.Tfs addin] to post issues as comments to pull requests.
+
+# OAuth authentication from Azure Pipelines
```
Comment (x-jokay) [inline docs/setup.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/setup.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+OAuth token provided to the build.
+For this you need to enable the [Allow scripts to access the OAuth token] option on the build definition.
+
+To authenticate you can use the [TfsAuthenticationOAuth] alias from the [Cake.Tfs addin].
+
+The user under which the build runs, named `<projectName> Build Service (<organizationName>)` (e.g. `Cake.Issues-Demo Build Service (cake-contrib)`),
+needs to have `Contribute to pull requests` permission for the specific repository to allow [Cake.Issues.PullRequests.Tfs addin]
+to post issues as comments to pull requests.
+
+# Azure Active Directory
```
Comment (x-jokay) [inline docs/setup.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/examples/azure-pipelines.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    if (isRunningOnAzureDevOps)
+        {
+            var repositoryUrl = new Uri(context.EnvironmentVariable("BUILD_REPOSITORY_URI"));
+
+            if (isPullRequestBuild)
+            {
+                if (!Int32.TryParse(context.EnvironmentVariable("SYSTEM_PULLREQUEST_PULLREQUESTID"), out var pullRequestId))
+                {
+                    throw new Exception(
+                        string.Format(
```
Comment (x-jokay) [inline docs/examples/azure-pipelines.md]: Add `CultureInfo.InvariantCulture`.

**Comment by pascalberger** [comment]
@x-jokay Ready for another review.

Top level headings are  required to have the entry appear in site TOC (See https://github.com/cake-contrib/Cake.Issues.Website/pull/41)

**Pull request merged**